Assets/__Scripts/MapEditor/Mapping/Selection/BoxSelectionPlacementController.cs

```diff
@@ -32,21 +32,56 @@ public class BoxSelectionPlacementController : PlacementController<MapEvent, Bea
 
     public override int PlacementXMax => int.MaxValue;
 
+    public Bounds bounds = default;
+    public CustomEventsContainer customCollection;
+    public EventsContainer eventsContainer;
+    public CreateEventTypeLabels labels;
+
+    // And I thought generics would make this method cleaner
+    private void TestForType<T, BO, BOC, BOCC>(RaycastHit hit, BeatmapObject.Type type) where T : PlacementController<BO, BOC, BOCC> where BO : BeatmapObject where BOC : BeatmapObjectContainer where BOCC : BeatmapObjectContainerCollection
```

Comment:
If you want to keep the generics clean, what you can do is make a non-generic `PlacementController` class, which would have all of the non-generic methods (if there even is any lmao), then have the generic `PlacementController<BO, BOC, BOCC>` class inherit that.

That, or you can just make the T argument `MonoBehaviour`, which for this case wouldn't even need the other 3 type arguments.